tests/test_dataapi.py

```diff
@@ -0,0 +1,138 @@
+import unittest
+import responses
+from learnosity_sdk.request import DataApi
+
+# This test uses the consumer key and secret for the demos consumer
+# this is the only consumer with publically available keys
+# Normally the consumer secret should not be committed to a public
+# repository like this one.
+security = {
+    'consumer_key': 'yis0TYCu7U9V4o7M',
+    'domain': 'demos.learnosity.com'
+}
+consumer_secret = '74c5fd430cf1242a527f6223aebd42d30464be22'
+request = {
+    # These items should already exist for the demos consumer
+    'references': ['item_2', 'item_3'],
+    'limit': 1
+}
+action = 'get'
+endpoint = 'https://data.learnosity.com/v1/itembank/items'
+
+
+class UnitTestDataApiClient(unittest.TestCase):
+    """
+    Tests to ensure that the Data API client functions correctly.
+    """
+
+    @responses.activate
+    def test_request(self):
+        """
+        Verify that `request` sends a request after it has been signed
+        """
+        dummy_response = {
+            'meta': {
+                'status': True,
+                'timestamp': 1514874527,
+                'records': 0
+            },
+            'data': []
+        }
+
+        responses.add(responses.POST, endpoint, json=dummy_response)
+        client = DataApi()
+        res = client.request(endpoint, security, consumer_secret, request,
+                             action)
+        assert res.json() == dummy_response
+        assert responses.calls[0].request.url == endpoint
+        assert 'signature' in responses.calls[0].request.body
+
+    @responses.activate
+    def test_request_iter(self):
+        """Verify that `request_iter` returns an iterator of pages"""
+        responses.add(responses.POST, endpoint, json={
+            'meta': {
```

Comment:
We could probably define those responses at the beginning of the file, as we reuse them multiple times, maybe just in an array, then do a 
```python
for r in responses:
    responses.add(responses.POST, endpoint, json=r)
```
here and in the test `results_iter` test below.

Probably also in `test_request` above, taking `responses[0].copy()`, overwriting `data` and removing `meta.next`